src/main/java/seedu/mtracker/model/subinstrument/Crypto.java

```diff
@@ -25,6 +28,25 @@ public String getRemark() {
         return remark;
     }
 
+    public void setExpiry(String inputExpiry) {
+        expiry = inputExpiry;
+    }
+
+    public void setRemark(String inputRemark) {
+        remark = inputRemark;
+    }
+
+    @Override
+    public void editParameter(HashMap<String,String> editedParameters) {
+        super.editParameter(editedParameters);
+        if (editedParameters.containsKey(EXPIRY_ATTRIBUTE)) {
+            setExpiry(editedParameters.get(EXPIRY_ATTRIBUTE));
+        }
+        if (editedParameters.containsKey(REMARK_ATTRIBUTE)) {
+            setRemark(editedParameters.get(REMARK_ATTRIBUTE));
+        }

```

Comment:
Same here abstract to individual methods. And create a editCryptoSpecificParameters. Similar to the addXYZCommand